synapse/lib/ast.py

```diff
@@ -3785,6 +3829,11 @@ async def callfunc(self, runt, argdefs, args, kwargs):
 
         opts = {'vars': mergargs}
 
+        if self.hasemit:
+            runt = await runt.getModRuntime(self.kids[2], opts=opts)
```

Comment:
It seems that in using getModRuntime instead of getSubRuntime we lose the outer scope, so a test like this fails for NoSuchVar on "x".

```
                $x=$lib.dict(foo=1)
                function foo() {
                    $mesg = (foo, $x.foo)
                    emit $mesg
                }
                $retn = $lib.list()
                for $y in $foo() { $retn.append($y) }
                return($retn)
```